source/jormungandr/jormungandr/interfaces/v1/EquipmentReports.py

```diff
@@ -68,18 +70,30 @@ def __init__(self):
     def options(self, **kwargs):
         return self.api_description(**kwargs)
 
-    def get(self, region=None, lon=None, lat=None, uri=None):
+    def _create_filter_equipment(self, instance):
+        code_types = [
+            code
+            for provider in instance.equipment_provider_manager._get_providers().values()
+            for code in provider.code_types
+        ]
+        if not code_types:
+            logging.getLogger(__name__).warn("No code type exists into equipment provider")
+            return "stop_point.has_code_type(no_code_types)"
+        filter = "stop_point.has_code_type("
+        for c in code_types:
+            filter = filter + str(c) + ","
+        return filter[:-1] + ")"
+
+    def get(self, region=None, lon=None, lat=None):
         self.region = i_manager.get_region(region, lon, lat)
         timezone.set_request_timezone(self.region)
         args = self.parsers["get"].parse_args()
+        instance = i_manager.instances.get(self.region)
 
-        uris = []
-        if uri:
-            if uri[-1] == "/":
-                uri = uri[:-1]
-            uris = uri.split("/")
-        args["filter"] = self.get_filter(uris, args)
-
+        # create filter
+        args["filter"] = self._create_filter_equipment(instance)
```

Comment:
You want to keep potential `filter` already defined by the user with something like:

```suggestion
        equipment_filter = self._create_filter_equipment(instance)
        if len(args["filter"]): args["filter"] += " and " + equipment_filter
        else : args["filter"] = equipment_filter
```

Reply:
I removed the filter option. 
But I think you've right, dispose of this option is cool if you can combine with the equipment_report filter. :+1:

Reply:
Last week I'll update with a integration test